### PR TITLE
CI: Switching to codecov

### PR DESCRIPTION
**Pull request by zssherman**
Coveralls is going to a pay model. Switch from coveralls to codecov.

### CODE REVIEW

**Comment by AdamTheisen** [comment]
@zssherman so we lost the test to ensure that the code coverage did not decrease by a certain threshold.  Do we have that ability with Codecov?

**Comment by zssherman** [comment]
Good point, I'll dig through the docs to see if I can find that

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  - 'setup.py'
+  - 'versioneer.py'
+
+coverage:
+  precision: 2
+  round: down
+  status:
+    project:
+      default:
+        target: 95
```
Comment (mgrover1) [inline codecov.yml:19]:
```suggestion
        target: 95
        threshold: 2%
```

**Review by mgrover1 — CHANGES_REQUESTED** [review]
One small suggestion - can we add in the "threshold" setting which allows up to a 2% decrease in code coverage in a PR?

**Comment by mgrover1** [comment]
@AdamTheisen @zssherman I suggested the configuration addition that sets the % decrease threshold

File: codecov.yml (hunk 1 above)
Comment (zssherman) [reply codecov.yml:19]: Ah perfect so there is a threshold! Changing now

Comment (mgrover1) [reply codecov.yml:19]: You betcha!

Comment (zssherman) [reply codecov.yml:19]: Has been changed

**Comment by AdamTheisen** [comment]
Thanks @mgrover1 !  How did I miss that.  I assume that the builds will fail if the coverage decreases right or is there a way to add a standalone check like we had with coveralls?

**Comment by mgrover1** [comment]
> Thanks @mgrover1 ! How did I miss that. I assume that the builds will fail if the coverage decreases right or is there a way to add a standalone check like we had with coveralls?

We would not be able to do this in a separate step/check.

**Review by mgrover1 — APPROVED** [review]

**Pull request merged**